python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -218,6 +252,19 @@ def trigger_out_of_band_tensor_transfer(
 
             src_actor = gpu_object_meta.src_actor
             tensor_meta = gpu_object_meta.tensor_meta
+            if gpu_object_meta.tensor_transport_backend == Backend.NIXL:
+                if src_actor == dst_actor:
```

Comment:
There is some logic later in this function that does the same thing based on src rank and dst rank but we should actually do this always no matter what the tensor transport is. Can you combine those checks and do it before this if block?

Reply:
But for nixl, it cannot get src_rank and dst_rank here.